src/Phoenix/PhoenixClient.ts

```diff
@@ -49,26 +51,34 @@ export class PhoenixClient {
 
   private async getContainerStatusAsync(containerData: IContainerData): Promise<ContainerInfo> {
     try {
-      const response = await window.fetch(
-        `${this.getPhoenixContainerPoolingEndPoint()}/${containerData.dbAccountName}/${containerData.forwardingId}`,
-        {
-          method: "GET",
-          headers: PhoenixClient.getHeaders(),
+      const runContainerStatusAsync = async () => {
+        const response = await window.fetch(
+          `${this.getPhoenixContainerPoolingEndPoint()}/${containerData.dbAccountName}/${containerData.forwardingId}`,
+          {
+            method: "GET",
+            headers: PhoenixClient.getHeaders(),
+          }
+        );
+        if (response.status === HttpStatusCodes.OK) {
+          const containerStatus = await response.json();
+          return {
+            durationLeftInMinutes: containerStatus?.durationLeftInMinutes,
+            notebookServerInfo: containerStatus?.notebookServerInfo,
+            status: ContainerStatusType.Active,
+          };
+        } else if (response.status === HttpStatusCodes.NotFound) {
+          throw new AbortError(response.statusText);
         }
-      );
-      if (response.status === HttpStatusCodes.OK) {
-        const containerStatus = await response.json();
-        return {
-          durationLeftInMinutes: containerStatus?.durationLeftInMinutes,
-          notebookServerInfo: containerStatus?.notebookServerInfo,
-          status: ContainerStatusType.Active,
-        };
-      }
-      return {
-        durationLeftInMinutes: undefined,
-        notebookServerInfo: undefined,
-        status: ContainerStatusType.Disconnected,
+        throw new Error(response.statusText);
       };
+      return (async () => {
+        return await promiseRetry(runContainerStatusAsync, {
+          onFailedAttempt: () => {
+            NotebookUtil.sleep(Notebook.retryAttemptDelayMs);
```

Comment:
promiseRetry in itself has a backoff mechanism, you dont have to sleep the thread explicitely. 

Use         
    
retryOptions = { forever: true, maxTimeout: this.retryIntervalInMs, minTimeout: this.retryIntervalInMs };
promiseRetry(runContainerStatusAsync, retryOptions);

This makes sure all retries are in the same interval. otherwise default is back off interval (2s, 4 s, 8s so on between retries)